File: java/src/Average.java
import java.util.Arrays;
import java.util.List;

public class Average {
    Rounding roundingMethod = new RoundDown();

    public void setRoundingMethod(Rounding roundingMethod){
        this.roundingMethod = roundingMethod;
    }

    public double calculateAverage(String numberAsString){
        List<String> numbers = Arrays.asList(numberAsString.split(","));
        Integer count = numbers.size();
        double total = 0;
        for (String number : numbers){
            total += Double.parseDouble(number);
        }
        return roundingMethod.round(total,count);
    }
}

File: python/src/Average.py
from python.src.Rounding import Rounding
from python.src.RoundUp import RoundUp
from python.src.RoundDown import RoundDown

class Average():
    roundingMethod = RoundDown()

    def setRoundingMethod(self,roundingMethod):
        self.roundingMethod = roundingMethod

    def calculateAverage(self,numbersAsString):
        numbers = numbersAsString.split(",")
        count = len(numbers)
        total = 0
        for number in numbers:
            total += int(number)
        average = self.roundingMethod.round(total,count)
        return average

    def main(self):
        numbers = input("Please input numbers e.g. 12,23,34")
        average = self.calculateAverage(numbers)
        print(average)

if __name__ == "__main__":
    average = Average()
    average.main()
